### PR TITLE
Feature/hero deprecation notice

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE-2460.

#### Changes proposed in this Pull Request:
- Add deprecation notice for `Hero` component changes coming in v6.0.0
- Upgrade storybook to RN 64

There's some irrelevant changes due to prettier... :/

### CODE REVIEW

File: components/src/core/hero/hero.tsx [hunk 1]
```diff
@@ -179,6 +193,46 @@ export const Hero: React.FC<HeroProps> = (props) => {
...
+        }
+    }, [valueColor]);
+    useEffect(() => {
+        if (units) {
+            // eslint-disable-next-line no-console
+            console.warn(
+                `Property 'units' in Hero component has been deprecated and will be removed in version 6.0.0. You should update to use the 'channelValueProps' prop instead.`
+            );
+        }
+    }, [units]);
```
Comment (joebochill) [inline components/src/core/hero/hero.tsx:235]: You have the deprecation warnings in here and they look good, but you also need to add the 'forward compatibility' so they can use the ChannelValueProps or the individual props in 5.4. That is, you'll need to include the changes that you made for 6.0.0 to introduce the ChannelValueProps prop, and then add some logic to either use the ChannelValueProps (if they're using it) or the individual props here in 5.4.

A very simple example for doing this with Icons can be seen in a few components like the EmptyState (https://github.com/pxblue/react-native-component-library/blob/dev/components/src/core/empty-state/empty-state.tsx#L108)

Essentially, you'll do something like:

```
const ChannelProps = props.ChannelValueProps || {icon: props.valueIcon, fontSize: props.fontSize, ...}
```
and then do the prop pass through just as you have it set up in 6.0.0.

Comment (joebochill) [reply components/src/core/hero/hero.tsx:235]: You'll also need to update the docs to show that the old props are deprecated and what the new prop is.

**Review by joebochill — APPROVED** [review]

**Comment by joebochill** [comment]
I approved, but you still need to update the docs/readme to add the strikethrough to the deprecated props and add a line for the ChannelValueProps.

**Comment by jeffvg** [comment]
@daileytj 
here is that error we talked about
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.

Execution failed for task ':app:processDebugManifest'.
> Manifest merger failed : uses-sdk:minSdkVersion 16 cannot be smaller than version 21 declared in library [com.facebook.react:react-native:0.64.2] /Users/e0458886/.gradle/caches/transforms-2/files-2.1/76d8faec94a3f038f7a3d609b4bd8e5e/jetified-react-native-0.64.2/AndroidManifest.xml as the library might be using APIs not available in 16
        Suggestion: use a compatible library with a minSdk of at most 16,
                or increase this project's minSdk version to at least 21,
                or use tools:overrideLibrary="com.facebook.react" to force usage (may lead to runtime failures)

**Pull request merged**